### PR TITLE
Merge executed transactions with multisend

### PR DESCRIPTION
**Pull request by fedgiac**
Many of our transactions can be merged together in the proposal creation interface.
This PR groups similar transactions together and executes them in batches.

Note that the Zodiac frontend will add the multisend contract by itself, so the output of the `deployment` script does _not_ contain multisend transactions, but just an array of blocks of transactions that will be merged manually in the interface.

Another thing to note is that there is no special case if the block contains a single transaction. This is also intended as this is also done by the Zodiac frontend when creating a new transaction.

### Test Plan

Modified unit tests. Also, test deployment on Rinkeby.

```
$ npx hardhat test-execute-proposal --network rinkeby --claims ./output/test-claims/claims.csv --settings ~/a/settings/file.json
Processing input files...
Generating Merkle proofs...
Clearing old files...
Saving generated data to file...
The following deterministic addresses will be deployed by the proposal:
cowDao: 0xBaDda00acA547DbC970e1f57372954F5d08DF0F7
teamController: 0xBAD7EAe6a77b8Cf80f1728439E275d6428Cf2362
investorFundsTarget: 0x3D92Ba95B8360F3aa6d44d08a6196fEBede48bc4
cowToken: 0xbaDCa1f0f4575c132A068b4054D4dD00c5E28996

Using executor 0x5415CF18e6aE326313F29AE4eE593fEf5595015e
Sent transaction 0x388d6df529f78b2ed433ececbaf3cf7c620f2c1b318f985c7c4d31c89330dfca
Sent transaction 0x1505b04fc7cd452ba89cccf6d285ecccfa5c84e823ed690a0ac5505f5d4afc8d
Sent transaction 0xcaccb16027214e0ff75b642f2189082ae7ee60deccedb6424be0ef8f9c0f0767
Sent transaction 0x88a527b13feb9a8e62e105ed88e22e71c36472aabc3911d2dd194dc5610055c5
```

See example multisent transactions.
- Multiple proxies created: https://rinkeby.etherscan.io/tx/0x388d6df529f78b2ed433ececbaf3cf7c620f2c1b318f985c7c4d31c89330dfca#eventlog
- All token transactions together: https://rinkeby.etherscan.io/tx/0x1505b04fc7cd452ba89cccf6d285ecccfa5c84e823ed690a0ac5505f5d4afc8d
(Note: the transfer to the bridge is missing as there is no bridge on Rinkeby, so the transfer wasn't triggered).

### CODE REVIEW

File: test/proposal.test.ts [hunk 1]
```diff
@@ -147,28 +147,27 @@ describe("proposal", function () {
...
+        );
       }
-      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-      expect(virtualTokenCreationTxResponse!).not.to.be.undefined;
 
-      const proxies = await contractsCreatedWithCreateCall(
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        virtualTokenCreationTxResponse!,
-        gnosisSafeManager.createCall.address,
+      const proxiesMinusKnownAddresses = allCreatedProxies.filter(
```
Comment (josojo) [inline test/proposal.test.ts:166]: Oh no, now you needed anyways to come up with this logic. Fortunately, it is not complex at all.

**Review by josojo — COMMENTED** [review]
Code looks good!

But I am not sure whether I am really a fan of it. It makes the debugging of the transactions harder, in realitymodule (maybe I am wrong here, have not investigated it a lot) and also later in etherscan.

But I am okay with merging if you think its a good idea. For sure, the execution later is easier!

**Review by josojo — APPROVED** [review]

**Comment by fedgiac** [comment]
I think the proposal looks better if it doesn't have too many transactions, but this is not too important and if you think 
There is a minor security concern on merging
```
      cowTokenCreationTransaction,
      approvalOmniBridgeTx,
      relayTestFundsToOmniBridgeTx,
      transferCowTokenToCowDao,
```
as the dao could take the new COW, but I don't really think the dao will do it.

>  It makes the debugging of the transactions harder, in realitymodule (maybe I am wrong here, have not investigated it a lot)

With "realitymodule" you mean the frontend? Note that the multisend is not used in the proposal creation, just in the execution. As it is right now, it will look like this:
https://snapshot.org/#/gnosis.eth/proposal/0x06feee3d2c9bc68ae0fccc8f2ada05c8323f0a71d2bcdfbc5556e8336cf5d71f
Note that there are batches with clearly labeled functions.

**Comment by josojo** [comment]
Let's go with this PR. Even if  the debugeablity on etherscan is not great, we can always go to tenderly

>Also note that the multisend will be added anyway even if the transactions are single transactions, or so I was told by Auryn.

I don't think so. Let's chat in sync.

**Pull request merged**